### ISSUE TITLE
[Feature]Add ability to persist GridSplitter position and restore it in the next session

### ISSUE THREAD

**Issue by dmitry007** [CommunityToolkit/Windows#276]
### Describe the problem

There is no easy way to query the current splitter position and restore it alter. There is no single Position property and no event firing when the splitter is moved. One can hook up resize event of one of the controls and then query row/column definitions for width/height, but GridSplitter does it anyway; need a way to expose it for external consumption.

When restoring the previously saved position, the logic in HorizontalMove / VerticalMove can be replicated externally, but this really needs to be exposed by the control itself.

### Describe the solution

Expose Position property and an event firing when it changes. 

### Help us help you

None

**Comment by dmitry007** [comment]
True, but that would involve copy-pasting the same code that GridSplitter already has in HorizontalMove / VerticalMove. Would be great to be able to re-use that code from outside to simplify the rehydration code.

**Comment by michael-hawker** [comment]
@dmitry007 `GridSplitter` isn't sealed, but I didn't say anything about OnDragHorizontal/OnDragVertical in there and why you're trying to add extra code there?

If you just hook into the `ManipulationCompleted` even already there on the GridSplitter you should be able to do anything you want when the sizer has been manipulated?

**Comment by dmitry007** [comment]
I am not trying to add any code to OnDragHorizontal/OnDragVertical - I am trying to reuse the code already there to be able to restore the splitter position, which would be easy to do. The control could store the *change* from the initial position (double _delta=0), with  OnDragHorizontal/OnDragVertical incrementing that field every time these methods are called (one can use separate counters for X and for Y). The value of the property can be persisted. When the value is restored, the Delta property setter can call OnDragHorizontal/OnDragVertical to restore the splitter position.
